### PR TITLE
Implementation of Hardy Ramanujan Algorithm in /maths

### PR DESCRIPTION
**Pull request by alfinwilliam**
Fixes: #1356

### CODE REVIEW

**Comment by cclauss** [comment]
Please add doctests as discussed in CONTRIBUTING.md.

**Comment by alfinwilliam** [comment]
Added docstrings & doctests as mentioned in CONTRIBUTING.md

File: maths/hardy_ramanujanalgo.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
+#It's a theorem that states the number of prime factors
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: #<space> please.

File: maths/hardy_ramanujanalgo.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# of n will approximately be log(log(n)) for most 
+#natural numbers n
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: #<space> please...

File: maths/hardy_ramanujanalgo.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import math 
+
+
+def exactPrimeFactorCount(n) : 
+	count = 0
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: Add a doctest?

File: maths/hardy_ramanujanalgo.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	if (n % 2 == 0) :
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: Remove the ()... This is Python, not JavaScript.

File: maths/hardy_ramanujanalgo.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+# of n will approximately be log(log(n)) for most 
+#natural numbers n
+
+import math 
+
+
+def exactPrimeFactorCount(n) : 
+	count = 0
+	if (n % 2 == 0) : 
+		count = count + 1
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: count += 1

File: maths/hardy_ramanujanalgo.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		while (n % 2 == 0) :
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: Remove the ()

File: maths/hardy_ramanujanalgo.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	if (n % 2 == 0) : 
+		count = count + 1
+		while (n % 2 == 0) : 
+			n = int(n / 2) 
+	#the n input value must be odd so that 
+	#we can skip one element ( ie i = i + 2)
+
+	i = 3
+	
+	while (i <= int(math.sqrt(n))) :
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: Remove the ()

File: maths/hardy_ramanujanalgo.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+		count = count + 1
+		while (n % 2 == 0) : 
+			n = int(n / 2) 
+	#the n input value must be odd so that 
+	#we can skip one element ( ie i = i + 2)
+
+	i = 3
+	
+	while (i <= int(math.sqrt(n))) : 
+		if (n % i == 0) :
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: Remove the ()

File: maths/hardy_ramanujanalgo.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	while (i <= int(math.sqrt(n))) : 
+		if (n % i == 0) :	 
+			count = count + 1
+			while (n % i == 0) : 
+				n = int(n / i) 
+		i = i + 2
+
+	#this condition checks the prime
+	#number n is greater than 2
+
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: Remove the ()

File: maths/hardy_ramanujanalgo.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+	if (n > 2) : 
+		count = count + 1
```
Comment (cclauss) [inline maths/hardy_ramanujanalgo.py]: +=

**Pull request merged**